get-artifacts/get_artifacts.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+function echoerr() {
+  cat <<< "$@" 1>&2;
+}
+
+#Check prerequists.
+function check_command() {
+    for var in "$@"; do
+      if ! (command -v "$var" >/dev/null 2>&1); then
+          echoerr "This command requires $var to be installed"
+          exit 1
+      fi
+    done
+}
+check_command jq curl zip echo getopts
+
+function usage() {
+  echoerr -t token used to communicate with github actions.
+  echoerr -h number of workflow_runs to pull artifacts from.
+  echoerr -a space seperated list of artifact names.
+  echoerr -b optional branch to pull historical artifacts from.
+  echoerr -w workflow file name of of workflow to pull artifact from.
+  echoerr -d target directory where subdirectories for jobs will be created and artifacts will be unzip to.
+  echoerr -i the single job to artifacts from.  Overrides -b,-w, and -h
+  echoerr -z decompress and delete the original downloaded artifacts?
+  echoerr -? this message.
+  echoerr output will written in to the -t target directory.
+}
```

Comment:
Typically help text is written to stdout, not stderr.

Reply:
I use standard error for any human intended output of a shell script/function, get_pr_info for example has its output sourced by the caller.   I don't know if it'll ever be the case that this scripts output is treated as input but have since viewed std error as logs for humans, and std out as return values for machines.   No machine will ever parse the help lines, or redirect for consumption.   It's far more likely that invalid input would cause the help method to be outputted.

Reply:
Hmm I don't think that division between stdout and stderr is standard. I think this tool should follow the standards used by other tools. I checked `cargo --help`, `rustc --help`, `git --help`, `curl --help`, `hg --help`: all of them output to standard out, not standard err.

In case of invalid input a different message should be printed to stderr, yes, and the exit code should be non-zero. I don't think the full help message needs to be printed to stderr (or stdout) in that case.